### PR TITLE
Check duplicate user when create or modify a user

### PR DESCRIPTION
**Pull request by hieuphq**
#### Status

READY

#### What's this PR do?

- [x] Change button text
- [x] Check duplicate user when create or modify

#### Where should the reviewer start?
/config/admin/admin.go
/config/auth/auth.go
/app/models/order.go
/app/models/shipper_order.go
/app/models/offer.go
/app/models/user.go

### CODE REVIEW

**Comment by hieuphq** [comment]
@ivkean help me to review this PR

File: app/model/user.go [hunk 1]
```diff
@@ -77,6 +78,15 @@ func (User) ConfigureQorResourceBeforeInitialize(resourcer resource.Resourcer) {
 
 		res.AddValidator(func(record interface{}, metaValues *resource.MetaValues, context *qor.Context) error {
 
+			db := context.GetDB()
+			username := ""
```
Comment (RuniVN) [inline app/model/user.go]: Use `var` when initializing zero value

File: config/auth/auth.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func (a Auth) GetCurrentUser(c *admin.Context) qor.CurrentUser {
 		}
 		user := User{}
 
-		err = a.DB.Table("users").Where("(username=? AND password=?) OR (email=? AND password=?)", value[0], value[1], value[0], value[1]).Find(&user).Error
+		err = a.DB.Table("users").Where("(LOWER(username)=LOWER(?) AND password=?) OR (LOWER(email)=LOWER(?) AND password=?)", value[0], value[1], value[0], value[1]).Find(&user).Error
 		if err != nil {
 			return nil
```
Comment (RuniVN) [inline config/auth/auth.go]: ?? why return nil?

Comment (hieuphq) [inline config/auth/auth.go]: @RuniVN return nil to prevent user can login to admin page

**Pull request merged**